File: 정수빈/LeetCode/easy/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):

    def maxDepth(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        self.maxH = 0

        def dfs(node, height):
            if node == None:
                return
            self.maxH = max(self.maxH, height)
            
            if node.left:
                dfs(node.left, height+1)
            if node.right:
                dfs(node.right, height+1)
        
        dfs(root, 1)
        return self.maxH
        

File: 장세훈/LeetCode/easy/MaximumDepthOfBinaryTree.java
class MaximumDepthOfBinaryTree {
    /*
        https://leetcode.com/problems/maximum-depth-of-binary-tree/submissions/1465826004/
        104. Maximum Depth of Binary Tree

        이진 트리의 최대 깊이를 구하는 문제
        dfs로 자식 노드로 이동하면서 깊이 카운팅
     */

    public int maxDepth(TreeNode root) {
        return dfs(root);
    }

    int dfs(TreeNode node) {
        if (node == null) return 0;
        return Math.max(dfs(node.right), dfs(node.left)) + 1;
    }
}